### PR TITLE
feat(proto): change tags from map to struct

### PR DESCRIPTION
**Pull request by hanxiao**

### CODE REVIEW

File: tests/unit/test_protobuf.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+class ProtobufTestCase(JinaTestCase):
+
+    def test_tags(self):
+        d = Document()
+        d.tags['int'] = 1  # will convert to float!!!
+        d.tags['str'] = 'blah'
+        d.tags['float'] = 0.1234
+        d.tags['bool'] = True
+        d.tags['nested'] = {'bool': True}
```
Comment (bwanglzu) [inline tests/unit/test_protobuf.py:16]: One suggestion is we can setup a set of test fixtures in tests and reuse these fixtures in other tests.

Reference:
> One great use for fixtures is to set up structures or variables the same way for all tests. This is to make sure that tests can run individually as well as a set and in any order. [link](https://pythontesting.net/framework/unittest/unittest-fixtures/)

Might be related to #692

File: tests/unit/test_protobuf.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+from jina.proto.jina_pb2 import Document
+from tests import JinaTestCase
+from google.protobuf.json_format import MessageToJson, Parse
+
+
+class ProtobufTestCase(JinaTestCase):
+
+    def test_tags(self):
+        d = Document()
+        d.tags['int'] = 1  # will convert to float!!!
```
Comment (hanxiao) [inline tests/unit/test_protobuf.py:12]: @JoanFM don't understand why but apparently this `Struct.Value` does not support `int`
 https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Value

**Comment by JoanFM** [comment]
@hanxiao, it is strange, I will look into it, but anyway it would be an improvement with respect with current tags right?

**Comment by JoanFM** [comment]
Found this stackoverflow conversation discussing the reason behind no ints. https://stackoverflow.com/questions/51818125/how-to-use-ints-in-a-protobuf-struct

**Comment by hanxiao** [comment]
- we can later make our own `Struct` by including `int32/64`.
- Note, the original `Struct` is designed as JSON compatible. In JS there is only `Number` no float or int. So when it comes to REST API, we may still have to actively parse float into int, e.g. `top_k`.

For now, we can merge it. Later we see how the current `Stuct` works for the doc enrichment and query lang and then decide how to refactor it.

**Pull request merged**